src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputFormattedTryCatchIfElse2.java

```diff
@@ -14,9 +14,12 @@ public static void main(String[] args) {
     try {
       /* foo */
     } catch (NullPointerException e) {
-      // violation below 'Empty catch block.'
+      // 2 violations 3 lines below:
+      //  ''}' at column 5 should be alone on a line.'
+      //  'Empty catch block.'
     } catch (Exception e) {
     } finally {
+      // violation above ''}' at column 5 should be alone on a line.'
```

Comment:
This is not expected violation, code is good and follows Google style.
Same for 2nd catch above from this line

Reply:
At least we clear that proposal was not good.
We need better proposal or we just need to abandon this effort.

It is ok to introduce false-positives by fixing false-negatives. 
Better to have false negatives, and mostly rely on fact that users use auto formatting or do not much care along fanatical following to style guide.

Reply:
https://checkstyle.org/property_types.html#RightCurlyOption

"alone" is not ok for LITERAL_CATCH.
But should be ok for "else" and "finally".

Reply:
@Zopsss , please create separate issue and PR to move "finally" to alone. It should always be alone.

Reply:
In this PR lets remove changes of rightfully and but let's keep update for suppression, let's see it it helps a bit. If behavior becomes better, we will merge small improvement.

Reply:
> It will not work LITERAL_ELSE, as there is
> 
> ```
> If () {
> } else if () {
> } else {
> }
> ```
> 
> @Zopsss , please create one more supplemental PR for this else-if , looks like we completely missed it. Sad that we do not see it such behavior diff.

test case added at https://github.com/checkstyle/checkstyle/pull/15785/files#r1807202504, I also added extra code to cover edge cases

Reply:
> please create separate issue and PR to move "finally" to alone. 

you suggested to keep updated suppression in this PR in your below comment ( https://github.com/checkstyle/checkstyle/pull/15779#discussion_r1805597135 ), I think there's no need of extra PR. We can update the suppression in this PR. Generating diff reports will also be easy

Reply:
After https://github.com/checkstyle/checkstyle/pull/15785 is merged, I will rebase this PR and we'll get more cases to test our suppression.

Reply:
That PR is merged

Reply:
lets rebase this PR and see what we have now.
I suggested split as I think that catch/else will not be fixed quickly, a bit more updates in xpath or .... will be required.
but "finally" looks like easy win, and we can merge it quicker, so this PR for catch/else will be smaller and might be easier to deal with.

Reply:
so in separate PR, we will only move ''finally'' to RightCurlyAlone and in this PR we will update the other pending suppression like moving ''catch'' and ''else'' to RightCurlyAlone?

Reply:
I have rebased the PR, one test was failing. Here's that test case: https://github.com/checkstyle/checkstyle/pull/15779/files#r1807376457. So now, what should I do in the separate PR? I'm little confuse about why we need separate PR. We can test everything here.

Reply:
I forgot to tell that I moved "CATCH" to "same". And kept "finally" in "alone".

Reply:
> I have rebased the PR, one test was failing

we need passed CI, please fix, only in this case we can see and trust violations in Inputs.

> We can test everything here

sure we can , but it will take you longer to make it. as you can see PR is passing in CI, so not ease to manage.
Problem is that we try to fix all handling at one PR, it will not be ease, but doable.
I do not see that we have good ideas for ELSE/CATCH tokens, they might not be fix-able that easy.
But more confidence in FINALLY.
While we continuing in this PR, you can create new PR to only move token FINALLY, it will be ease fix and quickly mergeable.
Up to you.

Reply:
let look at our goal as 3 independent updates:
- for LITERAL_ELSE
- for LITERAL_CATCH
- for LITERAL_FINALLY

lets have 3 PRs,

Reply:
PR sent for LITERAL_FINALLY  https://github.com/checkstyle/checkstyle/pull/15789

Reply:
At https://github.com/checkstyle/checkstyle/pull/15779#discussion_r1807474544, you suggested to keep CATCH/ELSE in "same" ( they're by currently in "same" so no need to change them ), and here you're suggesting that there will be 3 independent updates for CATCH/ELSE/FINALLY. I think we only need one extra PR which is for FINALLY.

Reply:
At https://github.com/checkstyle/checkstyle/pull/15779#discussion_r1807474544, you suggested to keep CATCH/ELSE in "same" and here you're suggesting that there will be 3 independent updates for CATCH/ELSE/FINALLY. I think we only need one extra PR which is for FINALLY.